node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentImpl.java

```diff
@@ -532,26 +533,32 @@ public void updateContainerNodeMetrics() {
         final long diskTotalBytes = (long) (nodeSpec.minDiskAvailableGb * BYTES_IN_GB);
         final Optional<Long> diskTotalBytesUsed = storageMaintainer.getDiskUsageFor(containerName);
 
+        lastCpuMetric.updateCpuDeltas(cpuSystemTotalTime, cpuContainerTotalTime, cpuContainerKernelTime);
+
         // CPU usage by a container as percentage of total host CPU, cpuPercentageOfHost, is given by dividing used
-        // CPU time by the container with CPU time used by the entire system.
+        // CPU time used by the container with CPU time used by the entire system.
+        double cpuUsageRatioOfHost = lastCpuMetric.getCpuUsageRatio();
+
         // CPU usage by a container as percentage of total CPU allocated to it is given by dividing the
         // cpuPercentageOfHost with the ratio of container minCpuCores by total number of CPU cores.
-        double cpuPercentageOfHost = lastCpuMetric.getCpuUsagePercentage(cpuContainerTotalTime, cpuSystemTotalTime);
-        double cpuPercentageOfAllocated = totalNumCpuCores * cpuPercentageOfHost / nodeSpec.minCpuCores;
+        double cpuUsageRatioOfAllocated = totalNumCpuCores * cpuUsageRatioOfHost / nodeSpec.minCpuCores;
+        double cpuKernelUsageRatioOfAllocated = cpuUsageRatioOfAllocated * lastCpuMetric.getCpuKernelUsageRatio();
```

Comment:
I would have expected: `double cpuKernelUsageRatioOfAllocated = totalNumCpuCores *  lastCpuMetric.getCpuKernelUsageRatio() / nodeSpec.minCpuCores *` Why use cpuUsageRatioOfHost?

Reply:
I took a shortcut, `getCpuKernelUsageRatio()` returned ratio of time that was spent in kernel mode of _non-idle CPU time_. Anyway, I see this is confusing now so I rearranged the terms so that they are calculated equally now.